File: backend/src/main/kotlin/com/github/plplmax/planning/subjects/Subject.kt
package com.github.plplmax.planning.subjects

import kotlinx.serialization.Serializable

@Serializable
data class Subject(
    val id: Int,
    val name: String,
    val complexity: Int,
    val minDaysBetween: Int,
    val minDaysStrict: Boolean,
    val onceFirstOrLastTimeslot: Boolean
)

File: frontend/src/subjects/Subject.ts
import type { Timeslot } from '@/timeslots/Timeslot'

export interface Subject {
  id: number
  name: string
  complexity: number
  minDaysBetween: number
  minDaysStrict: boolean
  onceFirstOrLastTimeslot: boolean
}

export interface SubjectDetail {
  id: number
  name: string
  complexity: number
  disallowedTimeslots: Timeslot[]
  minDaysBetween: number
  minDaysStrict: boolean
  onceFirstOrLastTimeslot: boolean
}
